extensions/positron-python/python_files/positron/positron_ipykernel/patch/bokeh.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+# Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+#
+
+from logging import Logger
+
+from ..session_mode import SessionMode
+
+MIME_TYPE_POSITRON_WEBVIEW_FLAG = "application/positron-webview-load.v0+json"
+
+
+def handle_bokeh_output(session_mode: SessionMode, logger: Logger) -> None:
+    """
+    Override the bokeh notebook display function to add a flag that the front-end can pick up on to
+    know that the data coming over should be replayed in multiple steps.
+
+    Args:
+        session_mode (SessionMode): The mode that the kernel application was started in.
+        logger (Callable[[str], None]): A logger function.
+    """
+    if session_mode == SessionMode.NOTEBOOK:
+        # Don't do anything if we're in a notebook
+        return
+
+    try:
+        from bokeh.io import notebook
+    except ImportError:
+        pass
+    else:
+        old_publish_display_data = getattr(notebook, "publish_display_data", None)
+
+        if old_publish_display_data is None:
+            logger.warning(
+                "Could not find bokeh.io.notebook.publish_display_data to update. Bokeh plots may not display correctly."
+            )
+            return
+
+        def new_publish_display_data(*args, **kwargs) -> None:
+            if isinstance(args[0], dict):
+                # Take the first arg, which is a dictionary and add a new key that will let the
+                # frontend know that this data comes from bokeh in notebook mode.
+                args[0][MIME_TYPE_POSITRON_WEBVIEW_FLAG] = ""
```

Comment:
Would this also appear in the eventual display message, and does that matter?

Reply:
It would and I dont think it matters. I feel like the logic to only target specific messages could get messy and delicate quickly. My assumption is that additional mime-types shouldnt ever break something.